rustls/src/server/hs.rs

```diff
@@ -65,41 +65,18 @@ pub fn can_resume(
     suite: &'static SupportedCipherSuite,
     sni: &Option<webpki::DnsName>,
     using_ems: bool,
-    resumedata: persist::ServerSessionValue,
-) -> Option<persist::ServerSessionValue> {
+    resumedata: &persist::ServerSessionValue,
+) -> bool {
     // The RFCs underspecify what happens if we try to resume to
     // an unoffered/varying suite.  We merely don't resume in weird cases.
     //
     // RFC 6066 says "A server that implements this extension MUST NOT accept
     // the request to resume the session if the server_name extension contains
     // a different name. Instead, it proceeds with a full handshake to
     // establish a new session."
-
-    if resumedata.cipher_suite == suite.suite
+    resumedata.cipher_suite == suite.suite
         && (resumedata.extended_ms == using_ems || (resumedata.extended_ms && !using_ems))
-        && same_dns_name_or_both_none(resumedata.sni.as_ref(), sni.as_ref())
-    {
-        return Some(resumedata);
-    }
-
-    None
-}
-
-// Require an exact match for the purpose of comparing SNI DNS Names from two
-// client hellos, even though a case-insensitive comparison might also be OK.
```

Comment:
To the extent webpki implements PartialEq for DNS names, it does so in a case-insensitive manner. Rustls was, prior this PR, using a case-sensitive comparison. If this is an intentional change then shouldn't we have a test that verifies that we're doing a case-insensitive comparison now? Otherwise, perhaps this aspect should be reverted? (If reverted, the implementation of this function could be made simpler.)

Reply:
The change was not intentional. I guess we should maybe revert this part.